src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,23 +179,20 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
-    boolean unfinishedEncapsulatedTokenRow = false;
+    boolean truncatedQuotedField = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
-      if (unfinishedEncapsulatedTokenRow) {
+      if (truncatedQuotedField) {
         onLastRow = true;
       }
 
-      try {
-        if (!onLastRow) {
-          record = iterator.next();
-        }
-      } catch (Exception e) {
+      if (!onLastRow) {
+        record = iterator.next();
```

Comment:
Right, so if you moved `iterator.next()` inside a `try` block and caught an "encapsulated token" exception, you would just want to set `onLastRow = true`, right? I don't see any reason to set `truncatedQuotedField` there. In fact, maybe we should just rename `truncatedQuotedField` to `onNextToLastRow` since the only action we take on it is to set `onLastRow = true` in the next loop iteration (or advance to the next record when we extend the stream)?